pkg/build/builder/sti.go

```diff
@@ -105,6 +105,9 @@ func (s *S2IBuilder) Build() error {
 	if runtimeImage != nil {
 		runtimeImageName = s.build.Spec.Strategy.SourceStrategy.RuntimeImage.Name
 		runtimeDockerAuth, _ = dockercfg.NewHelper().GetDockerAuth(runtimeImageName, dockercfg.PullAuthType)
+		// FIXME: do we need to explicitly turn off the incremental flag or make it
+		// part of the validation? or do we need to touch it at all?
+		s.build.Spec.Strategy.SourceStrategy.Incremental = false
```

Comment:
As far I understand it should be a part of validation logic. (S2I also checks this case but only when it's executed as a binary.)